src/applications/facility-locator/actions/index.js

```diff
@@ -275,6 +275,77 @@ export const genBBoxFromAddress = query => {
               placeName: features[0].place_name,
               placeType: features[0].place_type[0],
             },
+            searchArea: null,
+          },
+        });
+      })
+      .catch(_ => {
+        dispatch({ type: GEOCODE_FAILED });
+        dispatch({ type: SEARCH_FAILED, error: { type: 'mapBox' } });
+      });
+  };
+};
+
+/**
+ * Calculates a human readable location (address, zip, city, state) updated
+ * from the coordinates center of the map
+ */
+export const genSearchAreaFromCenter = query => {
+  const { lat, lng } = query;
+  return dispatch => {
+    const types = ['place', 'region', 'postcode', 'locality'];
+    mbxClient
+      .reverseGeocode({
+        countries: ['us', 'pr', 'ph', 'gu', 'as', 'mp'],
+        types,
+        query: [lng, lat],
+      })
+      .send()
+      .then(({ body: { features } }) => {
+        const coordinates = features[0].center;
+        const zip =
+          features[0].context.find(v => v.id.includes('postcode')) || {};
+        const location = zip.text || features[0].place_name;
+        const featureBox = features[0].box;
+
```

Comment:
What do you think about defining constants for each "corner" of the bounding box, and re-using them, like so?

```suggestion
const sw = coordinates[0] - BOUNDING_RADIUS;
const se = coordinates[1] - BOUNDING_RADIUS;
const nw = coordinates[0] + BOUNDING_RADIUS;
const ne = coordinates[1] + BOUNDING_RADIUS;

let minBounds = [sw, se, nw, ne];

if (featureBox) {
  minBounds = [
    Math.min(featureBox[0], sw),
    Math.min(featureBox[1], se),
    Math.max(featureBox[2], nw),
    Math.max(featureBox[3], ne),
  ];
}
```